### PR TITLE
Adding Azure Functions project to do stream processing logic

### PR DESCRIPTION
**Pull request by jmostella**
## Purpose

Adds the actual `Azure Function` that will do stream processing to the project

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Author pre-publish checklist:

- [x] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [ ] Made corresponding changes to the documentation

### CODE REVIEW

**Comment by jmostella** [comment]
Closes #102

File: e2e_samples/temperature_events/Functions/TemperatureEventsProj/host.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (NT-D) [inline e2e_samples/temperature_events/Functions/TemperatureEventsProj/host.json:1]: @jmostella Just FYI for future development.

When we host multiple functions in Function App (Azure resource), these functions utilize same host.json. It's OK if we use default host.json, but if we want to tweak some parameter for event-hub triggers, we need to consider 1) Prepare multiple Function App and enable only one function in each Function App 2) Overwrite host.json parameter by app settings like my repo.
https://github.com/NT-D/eh-func-iac/blob/main/terraform/function.tf#L49

Comment (jmostella) [reply e2e_samples/temperature_events/Functions/TemperatureEventsProj/host.json:1]: Yes I assumed that we would follow approach that was used in your blog

**Review by NT-D — APPROVED** [review]
LGTM

**Pull request merged**